diff --git a/src/slice.rs b/src/slice.rs
--- a/src/slice.rs
+++ b/src/slice.rs
@@ -1799,8 +1799,8 @@ impl<'a, 'b> std::cmp::PartialEq<RopeSlice<'b>> for RopeSlice<'a> {

         let mut chunk_itr_1 = self.chunks();
         let mut chunk_itr_2 = other.chunks();
-        let mut chunk1 = chunk_itr_1.next().unwrap_or("").as_bytes();
-        let mut chunk2 = chunk_itr_2.next().unwrap_or("").as_bytes();
+        let mut chunk1 = chunk_itr_1.next().unwrap_or("");
+        let mut chunk2 = chunk_itr_2.next().unwrap_or("");

         loop {
             if chunk1.len() > chunk2.len() {
@@ -1808,18 +1808,18 @@ impl<'a, 'b> std::cmp::PartialEq<RopeSlice<'b>> for RopeSlice<'a> {
                     return false;
                 } else {
                     chunk1 = &chunk1[chunk2.len()..];
-                    chunk2 = &[];
+                    chunk2 = "";
                 }
             } else if &chunk2[..chunk1.len()] != chunk1 {
                 return false;
             } else {
                 chunk2 = &chunk2[chunk1.len()..];
-                chunk1 = &[];
+                chunk1 = "";
             }

             if chunk1.is_empty() {
                 if let Some(chunk) = chunk_itr_1.next() {
-                    chunk1 = chunk.as_bytes();
+                    chunk1 = chunk;
                 } else {
                     break;
                 }
@@ -1827,7 +1827,7 @@ impl<'a, 'b> std::cmp::PartialEq<RopeSlice<'b>> for RopeSlice<'a> {

             if chunk2.is_empty() {
                 if let Some(chunk) = chunk_itr_2.next() {
-                    chunk2 = chunk.as_bytes();
+                    chunk2 = chunk;
                 } else {
                     break;
                 }
